include/power_grid_model/math_solver/math_solver.hpp

```diff
@@ -77,19 +106,39 @@ class MathSolver {
     void clear_solver() {
         newton_pf_solver_.reset();
         linear_pf_solver_.reset();
+        iterative_current_pf_solver_.reset();
+        linear_current_pf_solver_.reset();
     }
 
     void update_value(std::shared_ptr<MathModelParam<sym> const> const& math_model_param) {
         y_bus_.update_admittance(math_model_param);
+        // Invalidate prefactorization of iterative/linear current solvers
+        if (iterative_current_pf_solver_.has_value()) {
+            iterative_current_pf_solver_.value().reset_lhs();
+        }
+        if (linear_current_pf_solver_.has_value()) {
+            linear_current_pf_solver_.value().reset_lhs();
+        }
+    }
+
+    template <typename T>
+    MathOutput<sym> run_power_flow_iterative(T& base_solver, PowerFlowInput<sym> const& input, double err_tol,
```

Comment:
I don't understand why do we need to define a separate function here?
We can just call everything in `run_power_flow` right?